### ISSUE TITLE
throws an error if the extends resolves to a JS file

### ISSUE THREAD

**Issue by dylang** [privatenumber/get-tsconfig#7]
Hi!

[This code](https://github.com/privatenumber/get-tsconfig/blob/develop/src/utils/read-tsconfig.ts#L28-L39) looks for `require.resolve` result - and _only if that's not found_ does it try `path.join(extendsPath, 'tsconfig.json')`.

The problem is what if `require.resolve` resolves to an `index.ts` file. Then we try using that instead of `path.join(extendsPath, 'tsconfig.json')`. 

Then we try to read the JSON file. But it's JS not JSON, so we throw an error.

Example:

```json
"extends": "@company/tsconfig"
```

`@company/tsconfig` `package.json`:
```json
"main": "./src/index.ts"
```

`@company/tsconfig` `tsconfig.json`
```json
{
  // normal tsconfig.json
}
````

To resolve this we should do the resolving closer to how TypeScript does it. In this case, it would never resolve a .JS file, only .json. Perhaps if the filename is not `.json` we then attempt `path.join(extendsPath, 'tsconfig.json')`.

**Comment by privatenumber** [comment]
Is this a real issue you're encountering?

It currently throws if not a valid JSON:
https://github.com/privatenumber/get-tsconfig/blob/58a696ce9b775f1988b69e06410f40b48c975054/tests/specs/extends.spec.ts#L22-L34

**Comment by dylang** [comment]
Yes - it throws if it's not valid JSON, but what I'm hoping for is that it prioritizes using the `tsconfig.json` in that path instead of the `index.js`.

I'm running into this when using `tsx` in our monorepo. We have a company-standard `tsconfig` package that has `/tsconfig.json` and a `main` pointing to `/src/index.js`.

**Comment by privatenumber** [comment]
Gotcha.

Do you mind opening a PR with a failing test case?

**Comment by dylang** [comment]
Failing test created. 

BTW - the test framework throws an error but passes if the test JS doesn't compile. 😅

**Comment by privatenumber** [comment]
> BTW - the test framework throws an error but passes if the test JS doesn't compile. 😅

What does this mean?
